### PR TITLE
[#213] Support GHC-8.8

### LINKED ISSUE #213: Support GHC-8.8

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #213

Currently blocked by https://github.com/haskell-infra/hackage-trustees/issues/238

### CODE REVIEW

**Comment by chshersh** [comment]
@vrom911 Thanks for your work! Looks like nothing we can do here until the issue is resolved... Also, it looks like other TOML packages have become even more difficult to maintain. I've created an issue:

* https://github.com/kowainik/tomland/issues/215

**Comment by chshersh** [comment]
@vrom911 Looks like that issue was resolved!

File: cabal.project [hunk 1]
```diff
@@ -1,3 +1,5 @@
 packages: .
 
 allow-newer: htoml-megaparsec:unordered-containers
+allow-newer: toml-parser:base
+allow-newer: tasty-hspec:base
```
Comment (chshersh) [inline cabal.project]: I just checked and saw that `tasty-hspec` supports `base-4.13`. Maybe this line could be removed?

File: .travis.yml [hunk 2]
```diff
@@ -19,6 +19,7 @@ matrix:
   - ghc: 8.2.2
   - ghc: 8.4.4
   - ghc: 8.6.5
+  - ghc: 8.8.1
 
   # stack
   - ghc: 8.6.5
```
Comment (chshersh) [inline .travis.yml]: Maybe here also should be `ghc-8.8.1` since stack resolver is updated
```suggestion
   - ghc: 8.8.1
```

File: tomland.cabal [hunk 3]
```diff
@@ -54,6 +55,8 @@ common common-options
                        -Wpartial-fields
                        -fhide-source-paths
                        -freverse-errors
+  if impl(ghc >= 8.8.1)
+    ghc-options:       -Wmissing-deriving-strategies
```
Comment (chshersh) [inline tomland.cabal]: This is soooo awesome! Thanks for the refactoring and code improvements!

File: stack.yaml [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
+  - regex-tdfa-1.2.3.2  # for tasty-silver
   - toml-parser-0.1.0.0
 
   - htoml-megaparsec-2.1.0.3
   - composition-prelude-2.0.2.1  # needed for htoml-megaparsec
+
+allow-newer: true
+
+ghc-options:
+  "$locals": -Wmissing-deriving-strategies
```
Comment (chshersh) [inline stack.yaml]: Is this needed if we have it in common stanza? Or `stack` doesn't recognize it?

Comment (vrom911) [reply stack.yaml]: :thinking:, good question, I'm actually not sure about this. But I guess you're right, I'll remove it.

**Pull request merged**